elide-model-config/src/main/java/com/yahoo/elide/modelconfig/validator/DynamicConfigValidator.java

```diff
@@ -373,23 +470,34 @@ private static boolean validateTableConfig(ElideTableConfig elideTableConfig) {
             table.getDimensions().forEach(dim -> {
                 validateFieldNameUniqueness(tableFields, dim.getName(), table.getName());
                 validateSql(dim.getDefinition());
-                validateTableSource(elideTableConfig, dim.getTableSource());
+                validateTableSource(dim.getTableSource());
+                validateCheckExpr(dim.getReadAccess(), visitor);
             });
 
             table.getMeasures().forEach(measure -> {
                 validateFieldNameUniqueness(tableFields, measure.getName(), table.getName());
                 validateSql(measure.getDefinition());
+                validateCheckExpr(measure.getReadAccess(), visitor);
             });
 
             table.getJoins().forEach(join -> {
                 validateFieldNameUniqueness(tableFields, join.getName(), table.getName());
-                validateJoin(join, elideTableConfig);
+                validateJoin(join);
             });
+
+            validateCheckExpr(table.getReadAccess(), visitor);
         }
 
         return true;
     }
 
+    private void validateCheckExpr(String readAccess, PermissionExpressionVisitor visitor) {
+        if (!isNullOrEmpty(readAccess)) {
+            ParseTree root = EntityPermissions.parseExpression(readAccess);
+            visitor.visit(root).evaluate(null);
```

Comment:
This is probably not safe to evaluate checks like this.  Checks can do all kinds of crazy things (since they are developer provided code).

You can instead:
1.  Have the visitor extract a list/set of check names.
2. Lookup the check name either in the list of roles in the security.hjson file or check the dictionary (getCheckMappings)